dask/base.py

```diff
@@ -1605,4 +1608,51 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
+
+    Note: this may affect performance, since it will change the graph being
+    computed.
+
+    Parameters
+    ----------
+    msg : str
+          Format string for printing out value. Default "%s".
+          It's also possible to use placeholders compatible with str.format(),
+          such as "{}" or "{0}", or the named placeholder "{result}"
+    val : T
+          Lazy value, including a Dask collection, Scalar, etc.
```

Comment:
In the implementation you're explicitly asking for a dask collection.
```suggestion
          Any Dask collection
```